File: Valid Palindrome/validPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        alphanum_word = ""
        for char in s:
            if char.isalnum():
                alphanum_word = alphanum_word + char.lower()
        
        
        while(len(alphanum_word) > 0):
            if (alphanum_word[0] == alphanum_word[-1]):
                alphanum_word = alphanum_word[1:-1]
            else:
                return False
        
        return True

File: neetcode150/TwoPointers/validPalindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        const char* front = s.c_str();
        const char* back = front + s.length() - 1;

        while (front < back) {
            while (front < back && !isalnum(*front)) ++front;
            while (front < back && !isalnum(*back)) --back;

            if (front >= back) break;

            if (tolower(*front) != tolower(*back)) return false;
            cout << *front << " " << *back << "\n";
            ++front;
            --back;
        }

        return true;
    }
};
